src/mode/modeHandler.ts

```diff
@@ -1095,8 +1096,25 @@ export class ModeHandler implements vscode.Disposable {
         case ">":
           if (letterToTheLeft === "<") { return true; }
           break;
-        default:
-          return false;
+      }
+
+      // This section is mostly for vscode auto closing brackets without a user inserting them
+      const letterToTheRight = TextEditor.getLineAt(vimState.cursorPosition).text[vimState.cursorPosition.character];
+      if (letterToTheRight !== undefined) {
+        switch (key) {
+          case "{":
```

Comment:
Generally when there are giant switch statements that's a sign that you could better represent your stuff as a data structure. In this case, maybe something like this:

```
const pairings = { 
    "{" : "}",
    "[" : "]",
    "<" : ">",

    /* etc... */
}
```